kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -334,18 +358,36 @@ SetOfKeyValuePairs DBAdapter::lastReachableBlockDbUpdates(const SetOfKeyValuePai
     }
   }
 
+  // Split the keys: NonProvable and Provable Keys
+  SetOfKeyValuePairs non_provable_kvs;
+  SetOfKeyValuePairs provable_kvs = updates;
+  for (const auto &k : non_provable_key_set_) {
+    if (deletes.find(k) != std::cend(deletes)) {
+      throw std::runtime_error{"The deletes key set contains a non-provable key: " + k.toString()};
+    }
+    auto iter = provable_kvs.find(k);
+    if (iter != std::cend(provable_kvs)) {
+      non_provable_kvs.emplace(iter->first, iter->second);
```

Comment:
Based on https://en.cppreference.com/w/cpp/container/unordered_set/emplace, all iterators might be invalidated if rehashing occurs. I think that is problematic with the loop being done over `non_provable_key_set_`.

Reply:
I think above is a mistake, I didn't read the code correctly. Nothing is emplaced in `non_provable_key_set_`. Please ignore.